spec/factories.rb

```diff
@@ -149,124 +238,35 @@
-            ethnic_group: ethnicity.first,
-            ethnic_background: ethnicity.last,
-            disabilities: disabilities,
-            hesa_sex: hesa_sex,
-            hesa_disabilities: hesa_disabilities,
-            hesa_ethnicity: hesa_ethnicity,
-          }
-        end
-      end
-
-      trait :with_safeguarding_issues_disclosed do
-        safeguarding_issues_status { 'has_safeguarding_issues_to_declare' }
-        safeguarding_issues { 'I have a criminal conviction.' }
-      end
-
-      trait :with_no_safeguarding_issues_to_declare do
-        safeguarding_issues_status { 'no_safeguarding_issues_to_declare' }
-      end
-
-      trait :with_safeguarding_issues_never_asked do
-        safeguarding_issues_status { 'never_asked' }
-      end
-
-      trait :with_degree do
-        with_degree { true }
-      end
-
-      after(:build) do |application_form, evaluator|
-        if evaluator.with_gcses
-          create(:gcse_qualification, application_form: application_form, subject: 'maths')
-          create(:gcse_qualification, application_form: application_form, subject: 'english')
-          create(:gcse_qualification, application_form: application_form, subject: 'science')
-        end
+        if evaluator.full_work_history
+          current_year = Date.today.year
+          first_start_date = Faker::Date.in_date_period(year: current_year - 5)
+          first_end_date = Faker::Date.in_date_period(year: current_year - 4)
+          first_job = build(:application_work_experience, start_date: first_start_date, end_date: first_end_date)
 
-        if evaluator.with_degree
-          create(:degree_qualification, application_form: application_form)
-        end
+          second_start_date = Faker::Date.in_date_period(year: current_year - 3)
+          second_end_date = Faker::Date.between(from: 1.year.ago, to: 6.months.ago)
+          second_job = build(:application_work_experience, start_date: second_start_date, end_date: second_end_date)
 
-        create_list(:application_choice, evaluator.application_choices_count, application_form: application_form, status: 'unsubmitted')
-        create_list(:submitted_application_choice, evaluator.submitted_application_choices_count, application_form: application_form)
-        create_list(:reference, evaluator.references_count, evaluator.references_state, application_form: application_form)
-        # The application_form validates the length of this collection when
-        # it is created, which is BEFORE we create the references here.
-        # This then *caches* the association on the  application_form, and means
-        # you have to explicitly reload it to pick up the created references.
-        # We do this here, so we only have to do it in one place, rather than
-        # everywhere we refer to application_form.application_references in tests.
-        # See https://github.com/thoughtbot/factory_bot/issues/549 for details.
-        if evaluator.references_count > 0
```

Comment:
@cwrw

Reply:
spoke in person but for posterity, my question was:
I was just wondering around why `@application_form.application_references << FactoryBot.build(:reference, :not_requested_yet)` is preferred to `FactoryBot.create(:reference, :not_requested_yet, application_form: @application_form)` they seem to generate the same queries?
The thinking around this is to avoid having to reload the parent object, as appending objects to the parent object made anything created available to it as well, at least in the spec.